### PR TITLE
Set up mongodb memory server & supertest for testing

### LINKED ISSUE #83: Set up groundwork for testing backend API endpoints
## Description
 
We need to create tests for backend. To ensure that tests are robust and repeatable, we will use mock in-memory database and jest for testing. After that we need to create tests for #81.

## Acceptance Criteria

- [x] Set up mock database
- [x] Create an example usage for `person.controller.ts` under `src/controllers/__test__/person.controller.test.ts`
- [x] Document way to add tests

## Implementation details

- Add supertest and mongodb-memory-server as dependencies
- .env should include the following now:
Screen Shot 2022-03-10 at 12 21 54 AM
- During testing, express uses in-memory mongodb mock instead of the real mongodb. You can run test cases through `npm run test`
NOTE: your .env should now include `NODE_ENV=test`

## How does it all work together?
- supertest added to wrap express app during testing ('**mock server**')
- supertest is also used to make request to the **mock server**
- supertest requests can be chained to check response status 
eg) from `src/controllers/__test__/person.controller.test.ts`
Screen Shot 2022-03-10 at 12 32 26 AM

- mongodb-memory-server now set up so that it is available at port 27017 of 127.0.0.1 during testing ( can be configured at `src/test/config`)
- `src/test/db-handler` contains utils for creating, starting, cleaning, stopping mock database, and can be used by any test files for beforeAll/afterEach/afterAll
eg) from `src/controllers/__test__/person.controller.test.ts`
Screen Shot 2022-03-10 at 12 32 48 AM

 
- tests are organised within the `__test__` directory in each folder that the test subject belongs to. For instance, `person.controller.test.ts` is located at `src/controllers/__test__/person.controller.test.ts` and `person.controller.ts` is located at `src/controllers/__test__/person.controller.ts`

fix #83

### PR DESCRIPTION
**Pull request by hajineats**
NOTE: your .env should now include `NODE_ENV=test`

This is for creating integration tests for REST API endpoints. We create these integration tests under `src/routes/__test__`, with a naming convention of `<entity>.route.test.ts` as we are testing endpoints.

## How does it all work together?
- supertest added to wrap express app during testing ('**mock server**')
- supertest is also used to make request to the **mock server**
- supertest requests can be chained to check response status 
eg) from `src/controllers/__test__/person.controller.test.ts`
Screen Shot 2022-03-10 at 12 32 26 AM

- mongodb-memory-server now set up so that it is available at port 27017 of 127.0.0.1 during testing ( can be configured at `src/utils/test/config`)
- `src/utils/test/db-handler` contains utils for creating, starting, cleaning, stopping mock database, and can be used by any test files for beforeAll/afterEach/afterAll
eg) from `src/controllers/__test__/person.route.test.ts`
Screen Shot 2022-03-10 at 12 32 48 AM

- tests are organised within the `__test__` directory in each folder that the test subject belongs to. For instance, `person.routes.test.ts` is located at `src/routes/__test__/person.routes.test.ts` and `person.routes.ts` is located at `src/controllers/__test__/person.routes.ts`

fix #83

### CODE REVIEW

**Comment by sher812** [comment]
LGTM

**Review by sher812 — DISMISSED** [review]

**Review by jimwang6012 — APPROVED** [review]
LGTM

File: backend/README.md [hunk 1]
```diff
@@ -0,0 +1,93 @@
+# Forgettable - Backend
+
+This is the backend of Forgettable repository.
+
+## How to set up backend
+1. Make sure your current directory is `Forgettable/backend`
+2. Create .env under repo's `backend` folder
+    - `MONGODB_URI` should be set to a running mongodb instance url (should start with `mongodb+src://`)
+    - `PORT` should be set to 3001 (*that is, if proxy on `forgettable-frontend` is set to `http://localhost:3001/`)   
+    - `NODE_ENV` should be set to `test` if you are running `npm run test`
```
Comment (schi314) [inline backend/README.md:10]: From https://jestjs.io/docs/environment-variables, it says that JEST auto sets the NODE_ENV to `'test'` if it is not already. So we might not need to add this to `.env`?

Comment (hajineats) [reply backend/README.md:10]: ahhh I didn't know about this, thanks!

**Review by jimwang6012 — APPROVED** [review]
LGTM

**Review by schi314 — APPROVED** [review]
LGTM!

**Pull request merged**